networkx/algorithms/tests/test_dag.py

```diff
@@ -121,6 +122,52 @@ def test_descendants(self):
         assert_equal(descendants(G, 3), set())
         assert_raises(nx.NetworkXError, descendants, G, 8)
 
+    def test_transitive_closure(self):
+        G = nx.DiGraph([(1, 2), (2, 3), (3, 4)])
+        transitive_closure = nx.algorithms.dag.transitive_closure
+        assert_edges_equal(transitive_closure(G).edges(), [(1, 2), (1, 3),
+                                                     (1, 4), (2, 3), (2, 4),
+                                                     (3, 4)])
+        G = nx.DiGraph([(1, 2), (2, 3), (2, 4)])
+        assert_edges_equal(transitive_closure(G).edges(), [(1, 2), (1, 3),
+                                                     (1, 4), (2, 3), (2, 4)])
+        G = nx.Graph([(1, 2), (2, 3), (3, 4)])
+        assert_raises(nx.NetworkXNotImplemented, transitive_closure, G)
+
+    def _check_antichains(self, solution, result):
+        sol = [frozenset(a) for a in solution]
+        res = [frozenset(a) for a in result]
+        assert_true(set(sol) == set(res))
+
+    def test_antichains(self):
+        antichains = nx.algorithms.dag.antichains
+        G = nx.DiGraph([(1, 2), (2, 3), (3, 4)])
+        self._check_antichains(list(antichains(G)), [[], [4], [3], [2], [1]])
+        G = nx.DiGraph([(1, 2), (2, 3), (2, 4), (3, 5), (5, 6), (5, 7)])
+        self._check_antichains(list(antichains(G)), [[], [4], [7], [7, 4], [6],
+                                           [6, 4], [6, 7], [6, 7, 4],
+                                           [5], [5, 4], [3], [3, 4],
+                                           [2], [1]])
+        G = nx.DiGraph([(1, 2), (1, 3), (3, 4), (3, 5), (5, 6)])
+        self._check_antichains(list(antichains(G)), [[], [6], [5], [4], [4, 6],
+                                           [4, 5], [3], [2], [2, 6],
+                                           [2, 5], [2, 4], [2, 4, 6],
+                                           [2, 4, 5], [2, 3], [1]])
```

Comment:
This identation looks quite ugly and it's dificult to read. Could you use a local variable for each solution? Something like:

```python
G = nx.DiGraph([(1, 2), (1, 3), (3, 4), (3, 5), (5, 6)])
solution = [[], [4], [7], [7, 4], [6], [4, 5], [3], [2],
            [2, 6], [5], [5, 4], [3], [3, 4], [2], [1]]
self._check_antichains(list(antichains(G)), solution)
```